### PR TITLE
When avatar/header are GIF, generate static versions

### LINKED ISSUE #441: auto-playing gifs, gif avatars, and hover-to-play
this is a possibly niche issue, and not for the sake of performance, as i dont believe that even matters anymore, with such things.

as currently there are several ways to auto-play an animation, gif, small video of sorts, it doesnt matter, the issue has been mentioned that such things can actively be bad for some people, in regards to frightening them, or making it hard to focus.

as such, i propose an accessibility option in the settings to disable auto-playing media, and make it so that such things only autoplay when hovered over or clicked (or only play the first frame in all other cases)

EDIT: this can also help with short gifs that play quickly, and epilepsy. it's a serious matter to be considered

### LINKED ISSUE #596: Toggle gif avatar autoplay
It would be nice to have the ability to toggle gif avatar autoplay in the settings.

### PR DESCRIPTION
**Pull request by Gargron**
* Account API returns "avatar"/"avatar_static", "header"/"header_static"
* Static version is the same as original for other cases
* Web UI de-animates avatars in toots, lists of users (play on hover)

Fix #441, fix #596, prerequisite for #1064

### CODE REVIEW

File: app/models/account.rb [hunk 1]
```diff
@@ -12,12 +12,12 @@ class Account < ApplicationRecord
...
 
   # Avatar upload
-  has_attached_file :avatar, styles: { original: '120x120#' }, convert_options: { all: '-quality 80 -strip' }
+  has_attached_file :avatar, styles: ->(f) { avatar_styles(f) }, convert_options: { all: '-quality 80 -strip' }
   validates_attachment_content_type :avatar, content_type: IMAGE_MIME_TYPES
   validates_attachment_size :avatar, less_than: 2.megabytes
 
   # Header upload
-  has_attached_file :header, styles: { original: '700x335#' }, convert_options: { all: '-quality 80 -strip' }
+  has_attached_file :header, styles: ->(f) { header_styles(f) }, convert_options: { all: '-quality 80 -strip' }
```
Comment (krainboltgreene) [inline app/models/account.rb]: You can change this to 

``` ruby
has_attached_file :header, styles: method(:header_styles), convert_options: { all: '-quality 80 -strip' }
```

if you want.

**Review by krainboltgreene — APPROVED** [review]

**Comment by krainboltgreene** [comment]
What's up with these test failures?

**Comment by Gargron** [comment]
Oh, apparently we have some JS tests someone wrote a long time ago that are now failing because I changed the `<Avatar>` component

File: app/assets/javascripts/components/components/avatar.jsx [hunk 2]
```diff
@@ -1,103 +1,18 @@
...
     size: React.PropTypes.number.isRequired,
     style: React.PropTypes.object,
-    animated: React.PropTypes.bool
+    animate: React.PropTypes.bool
   },
 
   getDefaultProps () {
     return {
-      animated: true
+      animate: false
```
Comment (riking) [inline app/assets/javascripts/components/components/avatar.jsx]: I'm unfamiliar with React - is there an easy way for a user script to change this kind of thing?

Someone (not me) may want animated avatars even in non-hover. Would be nice to fetch that kind of thing from localStorage (or even the database, I guess.)

**Comment by Fastidious** [comment]
If this PR broke notifications (#1491), how in the world did the checks passed? Just curious.

**Comment by Gargron** [comment]
@Fastidious We don't have automated tests for the React.js front-end - only for the API and server in general. So front-end testing is done manually, and things can slip by.

**Pull request merged**